src/Evolution/BoundaryConditions/Type.hpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <iosfwd>
+
+namespace evolution::boundary_conditions {
+/*!
+ * \brief The type of boundary condition.
+ *
+ * There are generally two categories or types of boundary conditions. Those
+ * that:
+ *
+ * 1. set up additional cells or elements outside the computational domain
+ *    and then apply boundary corrections the same way that is done in the
+ *    the interior. For example, in a discontinuous Galerkin scheme these
+ *    would impose boundary conditions through a numerical flux boundary
+ *    correction term.
```

Comment:
It'd be helpful in this paragraph to use the word "ghost", since that is how you refer to these BCs further down (e.g., lines 24-28). I suggest perhaps at the beginning saying "set up additional cells or elements (ghost cells/elements) outside...", or adding a sentence in the middle of this paragraph "We call these ghost cells/elements".